### PR TITLE
initial logic for creating a segment speed data extract

### PR DESCRIPTION
**Pull request by kdiluca**
related to: #36 

----

using manila data to test:
example to run:
python create_data_extract.py /data/valhalla/osmlr-tiles/manila/0/002/415.osmlr 0 2415 segment_speeds

### CODE REVIEW

**Comment by kdiluca** [comment]
still need to update output directory to mimic the osmlr directory structure

**Comment by dnesbitt61** [comment]
looks good to me...do you have a sample output we can view?

**Comment by kdiluca** [comment]
level 0 - segment_speeds.gz 23.1MB 
level 1 - segment_speeds.gz 12.4MB 
                                               21.8MB
level 2 - segment_speeds.gz 87.6MB 
                                               51.9MB
                                               33.3MB
                                               20.3MB
                                               49.0MB

**Comment by kdiluca** [comment]
example for running level 2 tiles: 
python create_data_extract.py /data/valhalla/osmlr-tiles/manila/2/000/601/683.osmlr 2 000#601 683

File: scripts/create_data_extract.py [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+  return osmlr_tile
+
+
+#Iterates through the segments, listing the lengths
+def list_segment_lengths(osmlr_tile, level, tile_id):
+  id = 0
+  segments_dict = {}
+  for entry in osmlr_tile.entries:
+    length = 0
+    segment_id = (id << 25) | (tile_id << 3) | level
```
Comment (kdiluca) [inline scripts/create_data_extract.py]: also fixed tile id bug with using + instead of |

File: scripts/create_data_extract.py [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+    id += 1
+    if entry.HasField("segment"): 
+      for loc_ref in entry.segment.lrps:
+        if loc_ref.HasField('length'):
+          length = length + loc_ref.length
+        print segment_id, level, tile_id, id, length
```
Comment (kdiluca) [inline scripts/create_data_extract.py]: @dnesbitt61 , could something else be incorrect with how I'm getting the segment id from the manila data?  Or could something be wrong with the data that I'm using from your drive?  Lou is not seeing segment id 849766009720 in the 0/002/415.json data extract.

This is what I see at the expected index of 2532:

segmentId, level, tile, index, length
84926286712 0 2415 2532 92
84926286712 0 2415 2532 92
84959841144 0 2415 2533 263
84959841144 0 2415 2533 263
84993395576 0 2415 2534 263
84993395576 0 2415 2534 263

**Comment by drewda** [comment]
@kdiluca @kevinkreiser is anything in this PR still relevant? readme additions?

**Comment by kdiluca** [comment]
Possibly @drewda   Let me look back through the readme's that I had added for this branch and will update.

**Comment by kdiluca** [comment]
Actually @drewda, this isn't the branch that I thought it was.  This is irrelevant now since we did protobuf so I'll close.

**Pull request closed without merging**